packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -64,6 +65,7 @@ function DocSidebarItemCategory({
     }
     return isActive ? false : item.collapsed;
   });
+  const [initialLink, setInitialLink] = useState(undefined);
```

Comment:
We don't need state here, we need to read item and get the final link destination (the `to` props).

This should rather be useMemo than useState + useEffect, as it's a value that can be computed